frontend/app/src/types/pagination.ts

```diff
@@ -6,10 +6,10 @@ export interface TablePagination<T> {
 }
 
 export interface ApiPagination<T> {
-  limit: number;
-  offset: number;
-  orderByAttributes: (keyof T)[];
-  ascending: boolean[];
+  limit?: number;
+  offset?: number;
+  orderByAttributes?: (keyof T)[];
+  ascending?: boolean[];
```

Comment:
Why are all these properties turned optional?

Reply:
So remember the edit functionality, I need api to get asset data by identifier. It uses the same API, and treat `identifier`  as one of the filter. So actually I don't need pagination to get this data (I only need 1 data). So I make it optional.
Wdyt I can do better here?

Reply:
Instead of making it optional for all the places, we should better make the function we call take a `Partial<>` of this.

This way for the specific case all properties can be optional but for the other places, we keep them required.